src/os.cpp

```diff
@@ -1830,14 +1838,46 @@ Error os_file_open_r(Buf *full_path, OsFile *out_file) {
                     zig_unreachable();
                 case EACCES:
                     return ErrorAccess;
-                case EISDIR:
```

Comment:
`case EISDIR:` is dead code because it is only returned on files to be opened for writing:

```
[EISDIR]
    The named file is a directory,
    and the arguments specify that
    it is to be opened for writing.
```